httpcore/_backends/sync.py

```diff
@@ -42,16 +42,15 @@ def __exit__(
         self, exc_type: typing.Any, exc_val: typing.Any, exc_tb: typing.Any
     ) -> None:
         if self.timeout is not None:
-            assert self.timeout is not None
-            assert self._start is not None
-            self.timeout -= perf_counter() - self._start
-            if self.timeout == 0:  # pragma: no cover
-                # It is extremely unlikely, but it is possible that
-                # a socket operation will take exactly how long our
-                # timeout is; in such cases, we do not set timeout
-                # to 0 because 0 means there is no timeout at all,
-                # and we will raise an exception in a next operation.
-                self._expired = True
+            self._timeout -= perf_counter() - self._start
+
+    @property
+    def timeout(self):
+        return self._timeout
+
+    @timeout.setter
```

Comment:
Let's drop this setter. Because we do not provide timeout changing behavior, this setter appears to be overcomplicated.

Reply:
if it would not expose outside of class, we can drop it.

Reply:
Yes, let's do it.

Reply:
done, also dropped getter as we can direct use it.